docs/content/Details.md

```diff
@@ -1,71 +1,100 @@
+
+```jsx live
+<State>
+  {([]) => {
+    const {getDetailsProps} = useDetails({closeOnOutsideClick: true})
+    return (
+      <Details {...getDetailsProps()}>
+        <Button as="summary">Hello</Button>
+        This is some content
+      </Details>
+    )
+  }}
+</State>
 
-```jsx
-<Button as="summary">Summary text</Button>
 ```
 
-## With static children
+You can use the `open` state returned from the hook to conditionally render content:
 ```jsx live
-<Details>
-  <Button as="summary">Click me</Button>
-  <p>This should show and hide</p>
-</Details>
+<State>
+  {([]) => {
+    const {getDetailsProps, open} = useDetails({closeOnOutsideClick: true})
+    return (
+      <Details {...getDetailsProps()}>
+        <Button as="summary">{open ? 'open' : 'close'}</Button>
+        This is some content
+      </Details>
+    )
+  }}
+</State>
 
 ```
 
-## With children as a function
-The render function gets an object with the `open` render prop to allow you to conditionally update UI based on the open state of the dropdown:
+You can also manually show/hide the content using the `setOpen` function returned from the hook. This can be useful if you have action items in the content of the component such as confirmation buttons:
 
 ```jsx live
-<Details>
-  {({open}) => (
-    <>
-      <Button as="summary">
-        {open ? 'Hide' : 'Show'}
-      </Button>
-      <p>This should show and hide</p>
-    </>
-  )}
-</Details>
+<State>
+  {([]) => {
+    const {getDetailsProps, setOpen} = useDetails({closeOnOutsideClick: true})
+    return (
+      <Details {...getDetailsProps()}>
+        <Button as="summary">Delete item</Button>
+        Are you sure?
+        <ButtonDanger onClick={() => setOpen(false)}>Yes I'm sure</ButtonDanger>
+      </Details>
+    )
+  }}
+</State>
+
 ```
 
-## Manage the open state manually
-The `Details` element is built to also let you manage the open state and toggle functionality if necessary. Just provide values to the `open` and `onToggle` props.
+In previous versions of Primer React Components, we allowed users to pass in a custom `onToggle` function. You can do this now by overriding the `onToggle` function returned in `getDetailsProps`:
 
-**Note:** The `overlay` prop will not function automatically if you chose to provide your own `open` state. You'll need to implement this yourself. You can use the `onClickOutside` prop to implement and customize this behavior.
 
 ```jsx live
-<State default={false}>
-  {([open, setOpen]) => {
-    const handleToggle = (e) => setOpen(e.target.open)
-    const handleClickOutside = () => setOpen(false)
-
+<State>
+  {([]) => {
+    const {getDetailsProps, setOpen} = useDetails({closeOnOutsideClick: true})
     return (
-      <Details open={open} onToggle={handleToggle} onClickOutside={handleClickOutside} overlay>
-        <Button as="summary">Click me</Button>
-        <p>This should show and hide</p>
+      <Details {...getDetailsProps()} onToggle={() => window.alert('hi')}>
```

Comment:
I don't think that this actually works the way that we would want it to. The custom `onToggle` ends up being used instead of the one provided from the call to `getDetailsProps()`. It ends up leading to the `open` state not actually being changed because the `handleToggle` of the `useDetails` never gets called.

Here is a minimal repro of what I'm seeing: https://codesandbox.io/s/sad-albattani-ebspm?file=/src/App.tsx

The first component (which passes custom `onToggle` to the `<Details />` doesn't change the name of the button based on the `open` state, whereas the second one does.

The third component works because it calls the `onToggle` returned from `getDetailsProps()` _after_ calling its custom functionality.

This is definitely doable, but feels a little bit cumbersome now that I think about it.

We should probably at least clean up this example and we can think about ways to make this a little bit more user-friendly. One pattern I've seen is that the props getter (in this case `getDetailsProps` could take as a parameter an object whose keys match the values that it is going to spread. So instead of 

```
<Details {...getDetailsProps()} onToggle={() => window.alert('hi')}>
```
the API could be:
```
<Details {...getDetailsProps({onToggle: () => window.alert('hi')})} >
```

We don't need to make that change now though, I think that cleaning up the example is probably enough and then we can see if this is a pattern that gets used often.

Reply:
Updated the docs!